src/components/Mapviewer.tsx

```diff
@@ -1,161 +1,129 @@
+        />
+        {timestamps.map(timestamp => (
+          <WMSTileLayer
+            key={timestamp.time}
+            url={'/wms/'}
+            layers={'radar:5min'}
+            styles={'radar-5min'}
+            time={timestamp.time}
+            format={'image/png'}
+            uppercase={true}
+            bounds={getBounds(raster)}
+            tileSize={1024}
+            opacity={timestamp.time === timestamps[step].time ? 1 : 0}
+            onload={() => {
+              console.log('finish loading tiles for ', timestamp.time)
+              setTimestamps(timestamps.map(ts => {
+                if (ts.time === timestamp.time) {
+                  return {
+                    ...ts,
+                    loaded: true
+                  }
+                };
+                return ts;
+              }))
             }}
-            paint={{
-              // "circle-radius": 4,
-              // "circle-stroke-width": 1,
-              // "circle-stroke-color": 'grey',
-              // "circle-color": [
-              //   'case',
-              //   ['>', ["get", "object_id"], 1000],
-              //   'red',
-              //   'blue'
-              // ],
-              "text-color": [
-                'case',
-                ['>', ["get", "object_id"], 1000],
-                'blue',
-                'red'
-              ],
-              "icon-color": [
-                'case',
-                ['>', ["get", "object_id"], 1000],
-                'blue',
-                'red'
-              ],
-              // "icon-color": [
-              //   'match',
-              //   ["get", "object_name"],
-              //   "ZWOLLE",
-              //   'red',
-              //   "Dante",
-              //   'brown',
-              //   'blue'
-              // ]
+            // the onTileLoadStart happens when a tile is requested and starts loading
+            // in this case, stop the animation and set the "loaded" parameter back to false
+            // to indicate that the tiles are being reloaded
+            // However, it normally takes very long to reload all the tiles (way longer than the first time)
+            // During this time, user is not supposed to zoom in/zoom out, else the reloading process will happen again
+            ontileloadstart={() => {
+              stopAnimation()
+              setTimestamps(timestamps.map(ts => ({
+                ...ts,
+                loaded: false
+              })));
             }}
           />
-        </Source>
-      </ReactMapGL>
+        ))}
+      </Map>
+      <div>
+        <button
+          onClick={() => startAnimation()}
+          disabled={!tilesReady || playAnimation}
+        >
+          {tilesReady ? 'Play' : 'Loading ...'}
+        </button>
+        {' '}
+        <button
+          onClick={() => stopAnimation()}
```

Comment:
As stopAnimation is already a function with 0 arguments, this is the same as `onClick={stopAnimation}`, no need to create yet another function. (But onClick will send the event as first argument, so watch out with functions where that would be a problem). Same for startAnimation.

Reply:
noted